lambdas/format_input.py

```diff
@@ -0,0 +1,190 @@
+
+    # All required fields are present
+    if missing_fields := [field for field in REQUIRED_FIELDS if field not in event]:
+        raise ValueError(
+            f"Input must include all required fields. Missing fields: {missing_fields}"
+        )
+
+    # Valid starting step
+    starting_step = event["starting-step"]
+    if starting_step not in VALID_STARTING_STEPS:
+        raise ValueError(
+            f"Input 'starting-step' value must be one of: {VALID_STARTING_STEPS}. "
+            f"Value provided was '{starting_step}'"
+        )
+
+    # Valid harvest type
+    harvest_type = event["harvest-type"]
+    if harvest_type not in VALID_HARVEST_TYPES:
+        raise ValueError(
+            f"Input 'harvest-type' value must be one of: {VALID_HARVEST_TYPES}. Value "
+            f"provided was '{harvest_type}'"
+        )
+
+    # If starting with harvest step, required harvest fields are present
+    if starting_step == "harvest":
+        if missing_harvest_fields := [
+            field for field in REQUIRED_HARVEST_FIELDS if field not in event
+        ]:
+            raise ValueError(
+                "Input must include all required harvest fields when starting with "
+                f"harvest step. Missing fields: {missing_harvest_fields}"
+            )
+
+
+def generate_harvest_end_date_string(input_date: str) -> str:
+    input_date_object = None
+    for date_format in VALID_DATE_FORMATS:
+        try:
+            input_date_object = datetime.datetime.strptime(input_date, date_format)
+        except ValueError:
+            pass
+    if input_date_object:
+        harvest_end_date = input_date_object - datetime.timedelta(days=1)
+        return harvest_end_date.strftime("%Y-%m-%d")
+    raise ValueError(
+        "Input 'time' value must be one of the following date string formats: "
+        f"{VALID_DATE_FORMATS}. Value provided was '{input_date}'"
+    )
+
+
+def generate_load_input(
+    event: dict, harvest_type: str, transform_output_file: str
+) -> dict:
+    if prefix := event.get("index-prefix"):
+        index_name = f"{prefix}-{event['source']}"
+    else:
+        index_name = event["source"]
+
+    load_commands = [
+        f"--url={event['opensearch-url']}",
+        "ingest",
+        f"--source={index_name}",
+    ]
+
+    if harvest_type == "full":
+        load_commands.append("--new")
+        if prefix != "rdi":
+            load_commands.append("--auto")
+
+    load_commands.append(f"s3://{event['output-bucket']}/{transform_output_file}")
+
+    return {"commands": load_commands}
+
+
+def generate_transform_input(
+    event: dict, harvest_output_file: str, transform_output_file: str
+) -> dict:
+    transform_commands = [
+        f"--input-file=s3://{event['output-bucket']}/{harvest_output_file}",
+        f"--output-file=s3://{event['output-bucket']}/{transform_output_file}",
+        f"--source={event['source']}",
+    ]
+
+    if event.get("verbose") == "true".lower():
```

Comment:
Confirming, there's no `verbose` check need in `generate_load_input`?

Reply:
Correct, mario doesn't have a verbosity option in its cli. When we update this lambda to work with the new loader app we'll add a check since the new app will have that.